### ISSUE TITLE
[halo-pr-prep] Task 5: Update generation example script

### ISSUE THREAD

**Issue by jalengg** [jalengg/PyHealth#12]
## Task 5: Update generation example script

**Depends on:** Task 3 (needs refactored HALO model)

### Goal

Update \`examples/generate_synthetic_mimic3_halo.py\` to use PyHealth standard patterns for loading dataset and model checkpoint.

### Files to Modify

- **Modify:** \`examples/generate_synthetic_mimic3_halo.py\`

#### Step 1: Update generation script

Replace contents of \`examples/generate_synthetic_mimic3_halo.py\`:

**New pattern:**
\`\`\`python
import torch
import pickle
from pyhealth.datasets import MIMIC3Dataset
from pyhealth.tasks import halo_generation_mimic3_fn
from pyhealth.models.generators.halo import HALO

# STEP 1: Load dataset (needed for vocabulary)
base_dataset = MIMIC3Dataset(
    root=\"/srv/local/data/physionet.org/files/mimiciii/1.4\",
    tables=[\"DIAGNOSES_ICD\"],
    code_mapping={\"ICD9CM\": \"CCSCM\"},
    dev=False,
    refresh_cache=False,
)

# STEP 2: Set task
sample_dataset = base_dataset.set_task(halo_generation_mimic3_fn)

# STEP 3: Initialize model (same config as training)
model = HALO(
    dataset=sample_dataset,
    embed_dim=256,
    n_heads=4,
    n_layers=2,
    n_ctx=128,
    save_dir=\"./checkpoints/halo/\",
)

# STEP 4: Load trained checkpoint
checkpoint_path = \"./checkpoints/halo/halo_model\"
checkpoint = torch.load(checkpoint_path, map_location=model.device)
model.load_state_dict(checkpoint['model'])

# STEP 5: Generate synthetic patients
num_samples = 10000
synthetic_patients = model.synthesize_dataset(
    num_samples=num_samples,
    random_sampling=True
)

# STEP 6: Save synthetic data
output_path = \"./synthetic_data/halo_mimic3_10k.pkl\"
with open(output_path, 'wb') as f:
    pickle.dump(synthetic_patients, f)

print(f\"Generated {len(synthetic_patients)} synthetic patients\")
\`\`\`

#### Step 2: Test script

\`\`\`bash
python examples/generate_synthetic_mimic3_halo.py --help
\`\`\`
Expected: No import errors (needs trained checkpoint to fully execute)

#### Step 3: Commit

\`\`\`bash
git add examples/generate_synthetic_mimic3_halo.py
git commit -m \"refactor: update HALO generation example to use PyHealth patterns\"
\`\`\`

### Success Criteria

- [ ] Uses \`MIMIC3Dataset\` and \`halo_generation_mimic3_fn\`
- [ ] Uses refactored HALO model
- [ ] Loads checkpoint properly
- [ ] Calls \`model.synthesize_dataset()\`
- [ ] Saves synthetic data
- [ ] No import errors

### Related

- Plan: \`docs/plans/2026-02-16-halo-pyhealth-integration.md\` (lines 967-1056)
- Depends on: Task 3 (refactored HALO model)
- Label: **halo-pr-prep**

**Comment by jalengg** [comment]
Task 5 complete: examples/generate_synthetic_mimic3_halo.py has been rewritten for the PyHealth 2.0 API (commit 8892ddaf).

Old script used the deleted HALO_MIMIC3Dataset, raw pickle files for vocabulary, a MinimalDataset shim, and called HALOModel directly with manually assembled HALOConfig. It required --checkpoint and --output CLI args pointing at legacy pkl_data/ directories.

New script: loads MIMIC3Dataset + set_task(halo_generation_mimic3_fn) to reconstruct the vocabulary identically to training, instantiates HALO(dataset=sample_dataset, ...) with the same default hyperparameters as the training example, loads the checkpoint with torch.load + model.halo_model.load_state_dict(checkpoint['model']), calls model.synthesize_dataset(num_samples=1000, random_sampling=True), saves output as JSON (json.dump), and prints a summary of patients generated and average visits per patient. Optional --mimic3_root, --save_dir, --num_samples, and --output CLI args replace the mandatory legacy flags.

**Comment by jalengg** [comment]
Closing — Task 5 is complete and verified. The updated generation example passed code review (APPROVED with two minor comment suggestions applied in commit 4dcff7cf: table-mismatch warning and epochs-unused annotation). Verified clean in T9 final check. Commits: 8892ddaf, 4dcff7cf.
